alchemy/src/cloudflare/worker/miniflare-worker-options.ts

```diff
@@ -0,0 +1,480 @@
+import {
+  kCurrentWorker,
+  type MixedModeConnectionString,
+  type WorkerOptions,
+} from "miniflare";
+import assert from "node:assert";
+import { Self, type Binding, type WorkerBindingSpec } from "../bindings.ts";
+import type { WorkerProps } from "../worker.ts";
+
+export type MiniflareWorkerOptions = Pick<
+  WorkerProps,
+  | "bindings"
+  | "eventSources"
+  | "compatibilityDate"
+  | "compatibilityFlags"
+  | "format"
+> & {
+  name: string;
+  script: string;
+  remote: boolean;
+  port: number;
+};
+
+type BindingType = Exclude<Binding, string | Self>["type"];
+
+const REMOTE_ONLY_BINDING_TYPES = [
+  "ai",
+  "ai_gateway",
+  "browser",
+  "dispatch_namespace",
+  "vectorize",
+] satisfies BindingType[];
+const REMOTE_OPTIONAL_BINDING_TYPES = [
+  "d1",
+  "durable_object_namespace",
+  "images",
+  "kv_namespace",
+  "r2_bucket",
+  "queue",
+  "service",
+  "workflow",
+] satisfies BindingType[];
```

Comment:
What determines which buckets are remote only?

Reply:
Updated this so you specify on the resource.